stubs/urllib3/urllib3/util/url.pyi

```diff
@@ -1,13 +1,17 @@
-from typing import Any
-
 from .. import exceptions
 
 LocationParseError = exceptions.LocationParseError
 
 url_attrs: list[str]
 
 class Url:
-    slots: Any
+    auth: str | None
+    fragment: str | None
+    host: str | None
+    path: str | None
+    port: str | None
+    query: str | None
+    scheme: str | None
```

Comment:
At runtime, `Url` is a class that inherits from a `NamedTuple` that is also called `Url`. We can't mimic that in the stub exactly, but we can come pretty close by doing something like this:

```python
class _UrlBase(NamedTuple):
    auth: str | None
    fragment: str | None
    host: str | None
    path: str | None
    port: str | None
    query: str | None
    scheme: str | None

class Url(_UrlBase):
    # rest of the class definition as it is in the stub already
```

(GitHub won't let me use the "suggested changes" changes here because this line range includes some deleted lines.)

Reply:
very interesting, thanks for the suggestion 👍